### ISSUE TITLE
no network/internet access

### ISSUE THREAD

**Issue by fawaf** [microsoft/WSL#11675]
### Windows Version

Microsoft Windows [Version 10.0.22635.3720]

### WSL Version

0.0.0.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.153.1-microsoft-standard-WSL2

### Distro Version

Fedora

### Repro Steps

try to use any networking/internet command

### Expected Behavior

internet to be connected

### Actual Behavior

no internet/networking access

### Diagnostic Logs

[WslNetworkingLogs-2024-06-09_19-29-43.zip](https://github.com/user-attachments/files/15755839/WslNetworkingLogs-2024-06-09_19-29-43.zip)

**Comment by fawaf** [comment]
probably similar to https://github.com/microsoft/WSL/issues/11061

**Comment by fawaf** [comment]
also tried disabling all network interface as the issue linked about said, and that did not work also.

**Comment by dcasota** [comment]
WSL 2.2.4 has an issue, at least with an existing .wslconfig like the following:
```
# configure .wslconfig . If needed, change debugConsole=true
echo "[wsl2]">"$env:userprofile\.wslconfig"
echo "# Enable nested virtualization">>"$env:userprofile\.wslconfig"
echo "nestedVirtualization=true">>"$env:userprofile\.wslconfig"
echo "# networking mode">>"$env:userprofile\.wslconfig"
echo "networkingMode=NAT">>"$env:userprofile\.wslconfig"
echo "# Turn on default connection to bind WSL 2 localhost to Windows localhost">>"$env:userprofile\.wslconfig"
echo "localhostforwarding=true">>"$env:userprofile\.wslconfig"
echo "# turn off idle timeout">>"$env:userprofile\.wslconfig"
echo "vmIdleTimeout = -1">>"$env:userprofile\.wslconfig"
echo "# needed to start properly user-runtime-dir@0.service">>"$env:userprofile\.wslconfig"
echo "guiApplications = true">>"$env:userprofile\.wslconfig"
echo "# debug console" >>"$env:userprofile\.wslconfig"
echo "debugConsole=false">>"$env:userprofile\.wslconfig"
```

Downgrade to WSL 2.1.5 (https://github.com/microsoft/WSL/releases/download/2.1.5/wsl.2.1.5.0.x64.msi) resolves the issue.

Unfortunately I couldn't take the time to find out how to make run WSL 2.2.4 with the configuration mentioned.

**Comment by OneBlue** [comment]
@keith-horton: This user is using mirrored networking: 
```
networkingMode=mirrored
```

**Comment by fawaf** [comment]
yep, downgrading also did not work.

**Comment by fawaf** [comment]
also, i do not use any vpn on this particular box.

**Comment by cyberjj999** [comment]
> WSL 2.2.4 has an issue, at least with an existing .wslconfig like the following:
> 
> ```
> # configure .wslconfig . If needed, change debugConsole=true
> echo "[wsl2]">"$env:userprofile\.wslconfig"
> echo "# Enable nested virtualization">>"$env:userprofile\.wslconfig"
> echo "nestedVirtualization=true">>"$env:userprofile\.wslconfig"
> echo "# networking mode">>"$env:userprofile\.wslconfig"
> echo "networkingMode=NAT">>"$env:userprofile\.wslconfig"
> echo "# Turn on default connection to bind WSL 2 localhost to Windows localhost">>"$env:userprofile\.wslconfig"
> echo "localhostforwarding=true">>"$env:userprofile\.wslconfig"
> echo "# turn off idle timeout">>"$env:userprofile\.wslconfig"
> echo "vmIdleTimeout = -1">>"$env:userprofile\.wslconfig"
> echo "# needed to start properly user-runtime-dir@0.service">>"$env:userprofile\.wslconfig"
> echo "guiApplications = true">>"$env:userprofile\.wslconfig"
> echo "# debug console" >>"$env:userprofile\.wslconfig"
> echo "debugConsole=false">>"$env:userprofile\.wslconfig"
> ```
> 
> Downgrade to WSL 2.1.5 (https://github.com/microsoft/WSL/releases/download/2.1.5/wsl.2.1.5.0.x64.msi) resolves the issue.
> 
> Unfortunately I couldn't take the time to find out how to make run WSL 2.2.4 with the configuration mentioned.

Hi. I am facing the same network issue and couldn't resolve it for a while now.

Could you share how we can downgrade WSL from 2.2.4.0 to 2.1.5?

Do we just run the executable you link?

**Comment by dcasota** [comment]
Hi,
Case distinction:
- networkingMode=mirrored : I don‘t know the actual results neither with 2.1.5 nor with 2.2.4. As far as I remember, it didn’t work well in both versions (Windows 11 - patches - configuration). 
- networkingMode=NAT : In my lab, 2.1.5. works, 2.2.4. results in no network connectivity. Downgrade helps by simply starting the 2.1.5 .msi installer with local administrative privileges, click through and reopen a new Terminal(Administrator).

**Comment by cyberjj999** [comment]
Hi @dcasota . Thank you for your prompt response.

I ran the `.msi` installer and managed to downgrade my WSL - however, it seems like my network issues still persists.

For context, I do not have an existing `.wslconfig` in my `%USERProfile%` directory, and I believe the default WSL behavior is to use NAT

https://learn.microsoft.com/en-us/windows/wsl/networking
> Default networking mode: NAT
> By default, WSL uses a NAT (Network Address Translation) based architecture for networking. Keep the following considerations in mind when working with a NAT-based networking architecture:

Do you have some ideas of what other issues this network connectivity problem might be caused by?

Thanks so much.

**Comment by dcasota** [comment]
I tried to populate a curated answer. Hope this helps.

## Constellations

### 1a) enterprise workstation/laptop: Data Loss prevention policy
'Device control in Microsoft Defender for Endpoint' with an activated policy for W11 devices might detect WSL instances and prevent access to storage and networks.
In any case, prevent data loss. Ask company's IT compliance manager for help. Work together.

### 1b) Support good standing of IT security. Backup your system. 
   - Hardware bios/firmware level
   - Windows 11
   - Installed drivers and agents
   - Windows Subsystem for Linux
   - WSL Distribution
     For a WSL distribution, use `wsl --export <distribution-name> <path\filename.tar>`

### 2) WSL Reinstallation

Typically, a WSL reinstallation fixes misconfiguration issues quite well.

However, create first a backup of your WSL distribution as mentioned in Step 1. A reinstallation deletes any registered WSL distribution.

Here a sample of a reinstallation recipe.

```
# see https://learn.microsoft.com/en-us/windows/wsl/install-manual#step-1---enable-the-windows-subsystem-for-linux

# Open a Powershell Terminal (Administrator) 
dism /online /disable-feature /FeatureName:Microsoft-Windows-Subsystem-Linux
dism /online /disable-feature /featurename:VirtualMachinePlatform
dism /online /disable-feature /FeatureName:Microsoft-Hyper-V
rm "$env:userprofile\.wslconfig"

# reboot

# Open a Powershell Terminal (Administrator) 
dism /online /enable-Feature /All /FeatureName:Microsoft-Windows-Subsystem-Linux /norestart
dism /online /enable-feature /All /Featurename:VirtualMachinePlatform /norestart
dism /online /enable-Feature /All /FeatureName:Microsoft-Hyper-V /norestart
bcdedit /set hypervisorlaunchtype auto

# reboot

# Open a Powershell Terminal (Administrator) 
wsl --install
```


### 3) Network Connectivity

Network connectivity results as a sort of pipeline e.g. `physical network adapter ! virtual network adapter ! bridged ! vnic ! vswitch ! portgroup` .

If a vnic inside a WSL distribution does not get a network link, the pipeline is broken. Investigate if network connectivity works in another WSL distribution. If it works, it's rather not a WSL issue.

There are different types of network adapters e.g. built-in wired or wi-fi, usb(-c)-to-ethernet, etc. Not all vendor types are enabled by default in the Distros' Linux Kernel.

In case of a Windows 11 network configuration issue, check your configuration.
The WSL Networking Logs provided by @fawaf clearly testifies to the method required.

**Comment by CatalinFetoiu** [comment]
@fawaf thanks for reporting the issue

can you please collect a new set of networking logs using instructions below? - we need to the capture the process of WSL startup in the logs

run .\collect-networking-logs.ps1
run wsl --shutdown
start WSL and repro
stop .\collect-networking-logs.ps1

**Comment by fawaf** [comment]
oddly, i ran it, but it doesn't continue to collect logs. it just runs to completion and exits.

**Comment by fawaf** [comment]
a reinstall of wsl seems to have fixed this issue as mentioned above. i'll report back if it occurs again.

**Comment by cyberjj999** [comment]
> a reinstall of wsl seems to have fixed this issue as mentioned above. i'll report back if it occurs again.

Hey @fawaf - did you follow @dcasota's instruction above for reinstallation?

After reinstallation, how did you reimport your project files etc. back to your newly installed WSL2?

(I presume his Step 2 instructions above will delete your original WSL's Ubuntu content?)

**Comment by fawaf** [comment]
yep, that is correct. step 1 will back up the distro into a tar file.

i used the import command provided by wsl
```
wsl --import <Distro> <InstallLocation> <FileName>
```

**Comment by cyberjj999** [comment]
@fawaf Thank you for the swift replies!

Could I clarify that everything works well for you after re-importing?

i.e. your personal bash commands, project files, VSCode configuration, etc. etc. are all working ok?

^ just needed some re-assurance before I run the commands, lol!

**Comment by fawaf** [comment]
yep, everything works from what i can tell.

**Comment by dcasota** [comment]
@fawaf @cyberjj999 a question: can you check with `wmic qfe list` if you have a Windows Security Update KB5039212 installed?

There seems to be a build-in wsl update in KB5039212, because even after having uninstalled wsl + dism packages, `wsl -v` shows up as 2.2.4.

KB5039212 contains newer wsl bits:
```
wsl.exe,"10.0.22621.3672","07-Jun-2024","07:29","195,072"
wslconfig.exe,"10.0.22621.3672","07-Jun-2024","07:29","120,320"
wslg.exe,"10.0.22621.3672","07-Jun-2024","07:29","120,320"
```

Those wsl files are in c:\windows\system32.

I uninstalled the KB with `wusa /uninstall /kb:5039212`. The files weren't deleted, but `wsl -v` is back to 2.1.5.

in addition wsl show the well-known Error code `Wsl/Service/CreateInstance/GetDefaultDistro/WSL_E_DEFAULT_DISTRO_NOT_FOUND`. 

On this install base, a reinstallation succeed.

KB5039212 seems to already update the installbase to 2.2.4 despite issue(s) 'no network issue/internet access'.

**Comment by fawaf** [comment]
i do not have KB5039212 installed.

**Comment by cyberjj999** [comment]
> @fawaf @cyberjj999 a question: can you check with `wmic qfe list` if you have a Windows Security Update KB5039212 installed?
> 
> There seems to be a build-in wsl update in KB5039212, because even after having uninstalled wsl + dism packages, `wsl -v` shows up as 2.2.4.
> 
> KB5039212 contains newer wsl bits:
> 
> ```
> wsl.exe,"10.0.22621.3672","07-Jun-2024","07:29","195,072"
> wslconfig.exe,"10.0.22621.3672","07-Jun-2024","07:29","120,320"
> wslg.exe,"10.0.22621.3672","07-Jun-2024","07:29","120,320"
> ```
> 
> Those wsl files are in c:\windows\system32.
> 
> I uninstalled the KB with `wusa /uninstall /kb:5039212`. The files weren't deleted, but `wsl -v` is back to 2.1.5.
> 
> in addition wsl show the well-known Error code `Wsl/Service/CreateInstance/GetDefaultDistro/WSL_E_DEFAULT_DISTRO_NOT_FOUND`.
> 
> On this install base, a reinstallation succeeds.
> 
> KB5039212 seems to update the installbase to 2.2.4 despite issue(s) 'no network issue/internet access'. Placed an info at https://answers.microsoft.com/en-us/windows/forum/all/cumulative-updates-for-windows-june-2024/7e7ebdf9-4c8a-420e-aa6c-9ce470649cb3.

Yes. I do have `KB5039212` installed in my latest update - incidentally, it seems like the installation date is pretty near or maybe even coincide with the day I started facing the network issue...

is the fix to remove this update then?

**Comment by himpich** [comment]
I can confirm that uninstalling `KB5039212` (and reboot afterwards) help me with my problem (dns working, nat not)

**Comment by dcasota** [comment]
@cyberjj999 In my x86_64 lab, removing kb5039212 helped to afterwards reinstall wsl to 2.1.5 successfully and not have issues with networkingMode. Also, in Start > Settings > Windows Updates > choose pause updates e.g. for 7 days.

**Comment by dcasota** [comment]
This issue is not resolved.

wsl.2.4.10.0.x64.msi has still issues with mirrored and with NAT.

The wsl connectivity architecture results as a sort of pipeline e.g. `physical network adapter ! virtual network adapter ! bridged ! vnic ! vswitch ! portgroup ` for ipv4/ipv6 on x86_64/arm64.

Often seen issue constellations are with wireless and hotpluggable usb(-c)-2-ethernet multi-adapters. In those constellations the binding to wsl firewall vswitch does not work dynamically enough. In device manager consider all those extension/virtual adapters #2 #3 etc.. This 

There is no information that that architecture issue has been fixed with wsl.2.4.11.0.x64.msi. mirrored remains beta quality, I'm afraid.

wsl.2.1.5.0.x64.msi works issueless with NAT. But, this weekend after the latest updates, windows11 on a x86_64 laptop booted with no screen. 
What worked to fix the situation is:
- Uninstalling in F8 mode latest quality updates, and then,
- configuring in .wslconfig mirrored worked to have a starting wsl installation. And then,
- configuring in .wslconfig NAT and successfully start wsl.


```
wsl -v
WSL-Version: 2.1.5.0
Kernelversion: 5.15.146.1-2
WSLg-Version: 1.0.60
MSRDC-Version: 1.2.5105
Direct3D-Version: 1.611.1-81528511
DXCore-Version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows-Version: 10.0.26100.2033
```

```
wmic qfe list
Caption                                     CSName  Description      FixComments  HotFixID   InstallDate  InstalledBy          InstalledOn  Name  ServicePackInEffect  Status
http://support.microsoft.com/?kbid=5050577  xxxxx   Update                        KB5050577               NT-AUTORITÄT\SYSTEM  2/7/2025

https://support.microsoft.com/help/5044284  xxxxx   Security Update               KB5044284               NT-AUTORITÄT\SYSTEM  2/16/2025

                                            xxxxx   Security Update               KB5052085               NT-AUTORITÄT\SYSTEM  2/12/2025
```
